File: ios/TabLongPressedEvent.swift
import React


// RCTEvent is not defined for new arch.
protocol RCTEvent {}

@objc public class TabLongPressEvent: NSObject, RCTEvent {
  public var viewTag: NSNumber
  private var key: NSString
  public var coalescingKey: UInt16

  public var eventName: String {
    return "onTabLongPress"
  }

  @objc public init(reactTag: NSNumber, key: NSString, coalescingKey: UInt16) {
    self.viewTag = reactTag
    self.key = key
    self.coalescingKey = coalescingKey
    super.init()
  }

  public func canCoalesce() -> Bool {
    return false
  }

  public class func moduleDotMethod() -> String {
    return "RCTEventEmitter.receiveEvent"
  }

  public func arguments() -> [Any] {
    return [
      viewTag,
      RCTNormalizeInputEventName(eventName),
      [
        "key": key
      ]
    ]
  }
}

File: packages/react-native-bottom-tabs/android/src/main/java/com/rcttabview/events/TabLongPressedEvent.kt
package com.rcttabview.events

import com.facebook.react.bridge.Arguments
import com.facebook.react.uimanager.events.Event
import com.facebook.react.uimanager.events.RCTEventEmitter

class TabLongPressEvent(viewTag: Int, private val key: String) :
  Event<TabLongPressEvent>(viewTag) {

  companion object {
    const val EVENT_NAME = "onTabLongPress"
  }

  override fun getEventName(): String {
    return EVENT_NAME
  }

  override fun dispatch(rctEventEmitter: RCTEventEmitter) {
    val event = Arguments.createMap().apply {
      putString("key", key)
    }
    rctEventEmitter.receiveEvent(viewTag, eventName, event)
  }
}
